src/utils.ts

```diff
@@ -124,24 +129,32 @@ export const recursivelyDownloadTx = async (
 
   if (parsedTx.parents.length > 2) {
     // We downloaded a block, we should ignore it
-    return recursivelyDownloadTx(blockId, txIds, data);
+    return recursivelyDownloadTx(blockId, blockHeight, txIds, data);
   }
 
-  // If the first_block from the downloaded tx is not from the block we are searching, we should ignore it.
-  if (meta.first_block !== blockId) {
-    return recursivelyDownloadTx(blockId, txIds, data);
+  // Transaction is already confirmed
+  if (meta.first_block) {
```

Comment:
Maybe we can have here `if (meta.first_block && meta.first_block !== blockId)`?

In that case, we prevent doing this extra request to get the full block from the full node again.

Reply:
Done in 94ca49b33d78c9ec6d4c11251c6cfcaec94016ba

We wouldn't go to the fullnode there tho since we have a local cache and the block was already downloaded but it's nice to avoid it